core/src/main/java/com/netease/arctic/CommonUnifiedCatalog.java

```diff
@@ -42,8 +42,10 @@ public class CommonUnifiedCatalog implements UnifiedCatalog {
   private final Map<String, String> properties = Maps.newHashMap();
 
   public CommonUnifiedCatalog(
-      Supplier<CatalogMeta> catalogMetaSupplier, CatalogMeta meta, Map<String, String> properties) {
-    this.meta = meta;
+      Supplier<CatalogMeta> catalogMetaSupplier, Map<String, String> properties) {
+    CatalogMeta catalogMeta = catalogMetaSupplier.get();
+    CatalogUtil.mergeCatalogProperties(catalogMeta, properties);
+    this.meta = catalogMeta;
     this.properties.putAll(properties);
     this.metaSupplier = catalogMetaSupplier;
     initializeFormatCatalogs();
```

Comment:
Do you think this `initializeFormatCatalogs` method should be invoked when calling `com.netease.arctic.CommonUnifiedCatalog#refresh`?

Reply:
fixed